### PR TITLE
Eurotoken II - Trust score & unverified transfers

### PR DESCRIPTION
**Pull request by leondeklerk**
Stale branch of Euro ii team from spring 2022 class of Blockchain Engineering course.
This contribution directly conflicts with the merged changes by team "Euro iii". 

Adds unverified transfers, implementation on trust scores. Central gateway interaction still works.

### CODE REVIEW

**Comment by InvictusRMC** [comment]
Hi team please ensure your code passed our CI.

**Comment by InvictusRMC** [comment]
It has been over a week and your code is still not passing the CI. Please ensure your code is working before marking a PR as ready for review.

**Comment by leondeklerk** [comment]
@InvictusRMC we've fixed the CI issues.
Please also keep in mind that this PR is going to be merged into a separate branch, mainly just as a way to preserve our work as it cannot go into the main branch.

**Comment by devos50** [comment]
> it cannot go into the main branch.

Why can't it go in the main branch?

File: common/src/main/java/nl/tudelft/trustchain/common/eurotoken/TransactionRepository.kt [hunk 1]
```diff
@@ -669,21 +704,23 @@ class TransactionRepository(
...
-                    trustChainCommunity.sendBlock(
-                        trustChainCommunity.createAgreementBlock(
-                            block,
-                            block.transaction
-                        )
-                    )
-                }
-            }
-        )
+//        trustChainCommunity.registerBlockSigner(
```
Comment (InvictusRMC) [inline common/src/main/java/nl/tudelft/trustchain/common/eurotoken/TransactionRepository.kt:707]: Why is there commented code in your PR?

File: valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/dialogs/GatewayAddDialog.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            if (getGatewayStore().getGateways().isEmpty()) {
+                // If so set preferred to true and hide the switch
+                gateWayPreferredLayout.isVisible = false
+                gateWayPreferredSwitch.isChecked = true
+            }
+
+            connectGatewaySlider.onSlideCompleteListener = object : SlideToActView.OnSlideCompleteListener {
+                override fun onSlideComplete(view: SlideToActView) {
+
+                    @Suppress("DEPRECATION")
```
Comment (InvictusRMC) [inline valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/dialogs/GatewayAddDialog.kt:63]: Add a comment explaining why deprecated code is required here.

File: valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/dialogs/GatewayWarningDialog.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import androidx.annotation.RequiresApi
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import nl.tudelft.trustchain.valuetransfer.R
+import nl.tudelft.trustchain.valuetransfer.ui.VTDialogFragment
+import nl.tudelft.trustchain.valuetransfer.util.*
+
+class GatewayWarningDialog : VTDialogFragment() {
+
+    @RequiresApi(Build.VERSION_CODES.O)
```
Comment (InvictusRMC) [inline valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/dialogs/GatewayWarningDialog.kt:15]: What happens on older devices?

File: valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/passport/PassportCaptureActivity.kt [hunk 4]
```diff
@@ -48,6 +50,27 @@ open class PassportCaptureActivity : Activity(), TextRecognitionProcessor.Result
             onBackPressed()
         }
 
+        // Override the scanning
+        findViewById<TextView>(R.id.tvScanTitle).setOnClickListener {
+            overrideScan += 1
+            if (overrideScan % 5 == 0) {
+                onSuccess(
+                    MRZInfo(
```
Comment (InvictusRMC) [inline valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/passport/PassportCaptureActivity.kt:58]: Why is this placeholder data here?

File: valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/ui/trust/TrustFragment.kt [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+        lifecycleScope.launchWhenStarted {
+            while (isActive) {
+                refreshTransactions()
+
+                delay(1000)
+            }
+        }
+    }
+
+    @SuppressLint("RestrictedApi")
```
Comment (InvictusRMC) [inline valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/ui/trust/TrustFragment.kt:51]: Why is this suppressed?

**Review by InvictusRMC — CHANGES_REQUESTED** [review]
Code is looking good, with some minor comments.

Also team, please respond to @devos50's question.

**Comment by leondeklerk** [comment]
@devos50 this can't be merged with main as it directly conflicts with the work of Eurotoken III and we decided with @synctext to put our contribution on a separate branch to preserve the work in this repository.

File: common/src/main/java/nl/tudelft/trustchain/common/eurotoken/TransactionRepository.kt (hunk 1 above)
Comment (leondeklerk) [reply common/src/main/java/nl/tudelft/trustchain/common/eurotoken/TransactionRepository.kt]: Removed comment

File: valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/dialogs/GatewayAddDialog.kt (hunk 2 above)
Comment (leondeklerk) [reply valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/dialogs/GatewayAddDialog.kt]: Refactored to use not deprecated code

File: valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/passport/PassportCaptureActivity.kt (hunk 4 above)
Comment (leondeklerk) [reply valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/passport/PassportCaptureActivity.kt:58]: This is used to override the need to scan a passport or ID and inserts dummy data. As when working on an emulator it is quite difficult to capture a picture. 
This also complements the override already available to skip the NFC check.

File: valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/dialogs/GatewayWarningDialog.kt (hunk 3 above)
Comment (leondeklerk) [reply valuetransfer/src/main/java/nl/tudelft/trustchain/valuetransfer/dialogs/GatewayWarningDialog.kt]: We've checked and this check does not give any warnings so it has been removed.

**Comment by leondeklerk** [comment]
As for the failing CI, that seems to be an issue with https://raw.githubusercontent.com/blockchain/api-v1-client-java/mvn-repo/ not existing (anymore?) Which is not related to any of our changes.

**Comment by devos50** [comment]
Looks like [their entire repository](https://github.com/blockchain/api-v1-client-java) got removed. Seems to be the risk of relying on third-party software...

We'll look into it.

**Pull request merged**